axonserver/src/main/java/io/axoniq/axonserver/localstorage/file/SegmentBasedEventStore.java

```diff
@@ -469,26 +472,44 @@ private int retrieveEventsForAnAggregate(long segment, List<Integer> indexEntrie
     }
 
     @Override
-    public void health(Health.Builder builder) {
+    public void health(Health.Builder builder, DiskSpaceHealthIndicatorProperties diskSpaceHealthProperties) {
         String storage = storageProperties.getStorage(context);
         SegmentBasedEventStore n = next;
         Path path = Paths.get(storage);
         try {
             FileStore store = Files.getFileStore(path);
+
+            long diskFreeInBytes = store.getUsableSpace();
+            long threshold = diskSpaceHealthProperties.getThreshold().toBytes();
+            if (store.getUsableSpace() >= threshold) {
+                builder.up();
+            } else {
+                logger.warn(String.format("Free disk space below threshold. Available: %d bytes (threshold: %s)", diskFreeInBytes, threshold));
+                builder.status(new Status("WARN"));
+            }
+
+            builder.withDetail(context + ".total", store.getTotalSpace());
             builder.withDetail(context + ".free", store.getUsableSpace());
             builder.withDetail(context + ".path", path.toString());
+            builder.withDetail(context + ".threshold", threshold);
         } catch (IOException e) {
             logger.warn("Failed to retrieve filestore for {}", path, e);
         }
         while (n != null) {
```

Comment:
this while loop can be removed, was part of an initial implementation of multi-location storage that is no longer implemented.